### ISSUE TITLE
The format reward decline to zero!

### ISSUE THREAD

**Issue by Hasuer** [huggingface/open-r1#363]
I have done GRPO on Qwen2.5-1.5B-Instruct, as the base model, and use NuminaMath-TIR as the training dataset for one epoch.

> def format_reward(completions, **kwargs):
    """Reward function that checks if the completion has a specific format."""
    pattern = r"^<think>.*?</think>\s*<answer>.*?</answer>$"
    completion_contents = [completion[0]["content"] for completion in completions]
    matches = [re.match(pattern, content, re.DOTALL | re.MULTILINE) for content in completion_contents]
    return [1.0 if match else 0.0 for match in matches]

<img width="1635" alt="Image" src="https://github.com/user-attachments/assets/27db43de-11c1-49f6-8dcb-0b0bde8a1fed" />

Does anyone knows why the format reward decline to zero after ~50 steps? 

Besides, the performace no gpqa:diamond decline from 0.3081(base mode) to 0.2424 (RL model).

**Comment by Hasuer** [comment]
<img width="1518" alt="Image" src="https://github.com/user-attachments/assets/8e8b0a0f-8bab-49ae-9c12-25233ee4d058" />

It seems like the model output length exceeds "max_completion_length: 1024", and the regex cannot extract the right pattern

**Comment by Hasuer** [comment]
The dataset I used for training is **NuminaMath-TIR** for just one epoch.

Here is the config file:
# Model arguments
model_name_or_path: Qwen/Qwen2.5-1.5B-Instruct
model_revision: main
torch_dtype: bfloat16
attn_implementation: flash_attention_2

# Data training arguments
dataset_name: AI-MO/NuminaMath-TIR
dataset_configs:
- all
# Num processes is less by 1 as vLLM is using 1 GPU
num_processes: 7

# GRPO trainer config
bf16: true
use_vllm: true
vllm_device: auto
vllm_gpu_memory_utilization: 0.7
do_eval: true
eval_strategy: steps
eval_steps: 100
gradient_accumulation_steps: 16
gradient_checkpointing: true
gradient_checkpointing_kwargs:
  use_reentrant: false
hub_model_id: Qwen2.5-1.5B-Open-R1-GRPO
hub_strategy: every_save
learning_rate: 2.0e-05
log_level: info
logging_steps: 5
logging_strategy: steps
lr_scheduler_type: cosine
max_prompt_length: 512
max_completion_length: 1024
max_steps: -1
num_generations: 7
num_train_epochs: 1
output_dir: data/Qwen2.5-1.5B-Open-R1-GRPO
overwrite_output_dir: true
per_device_eval_batch_size: 32
per_device_train_batch_size: 16
push_to_hub: true
report_to:
- wandb
save_strategy: "no"
seed: 42
warmup_ratio: 0.1

**Comment by Hasuer** [comment]
The reason is the acc reward func cannot extract the answer as the model is a base (unaligned) model. which can't put the answer in \boxed{}.

**Comment by AXy1527** [comment]
> 基

Do you solve it? How to solve this problem.

**Comment by Hasuer** [comment]
> Do you solve it? How to solve this problem.

I use Qwen2.5-Math-7B as the base model and use NuminaMath-TIR as the training dataset for one epoch.

Seems worked.

<img width="1203" alt="Image" src="https://github.com/user-attachments/assets/7887a3d1-9ecc-4ec9-b4c2-0341a65b4000" />
